### PR TITLE
NettingChannelLibrary simplification

### PR DESCRIPTION
**Pull request by hrishikeshio**
Solves #437

### CODE REVIEW

**Comment by LefterisJP** [comment]
Thank you for the PR @hrishikeshio. We will review it but before anything else, you seem to have 12 commits for a 25 line diff. Can you please squash them all into one commit?

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 1]
```diff
@@ -633,6 +616,20 @@ library NettingChannelLibrary {
         }
     }
 
+    function index_or_throw(Data storage self, address caller_address) private returns (uint8 n) {
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: I think that the point of issue #437 is not to abstract away the logic into an `index_or_throw()` function. Instead I believe the point is to use a mapping, from participant address to index, which would be initialized in the constructor of the Netting channel and then can be queried [here](https://github.com/raiden-network/raiden/pull/480/files#diff-09cebda618db5fb2268a55de8de94279R318).

So the implementation of `index_or_throw()` should be a simple mapping lookup, that will throw if the lookup fails.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 2]
```diff
@@ -633,6 +616,20 @@ library NettingChannelLibrary {
...
+        // Return index of participant, or throw
+        n=0;
+        Participant storage participant = self.participants[0];
+        if (participant.node_address != caller_address) {
+            n=1;
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: just a style note: spaces around operators, in this case `=`

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 3]
```diff
@@ -326,15 +313,11 @@ library NettingChannelLibrary {
 
         if (hashlock != sha3(secret))
             throw;
+        
+        //Check if caller_address is a participant and select her partner
+        uint8 index = index_or_throw(self, caller_address)==0 ? 1 : 0;
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: just a style note: spaces around operators, in this case `==`

File: raiden/smart_contracts/NettingChannelLibrary.sol (hunk 1 above)
Comment (hrishikeshio) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: I have made the changes. Please review.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 4]
```diff
@@ -326,15 +314,11 @@ library NettingChannelLibrary {
...
-        Participant storage participant = participants[0];
-        if (participant.node_address == caller_address) {
-            participant = participants[1];
-            if (participant.node_address != caller_address) {
-                throw;
-            }
-        }
+        
+        //Check if caller_address is a participant and select her partner
+        uint8 index = 1 - index_or_throw(self, caller_address);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Looks nice here. But can you perhaprs rename the `participant` variable in this functon as `partner`?

Comment (hrishikeshio) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Made the changes @LefterisJP

**Pull request merged**